File: go/2move-zeros/283.移动零.go
/*
 * @lc app=leetcode.cn id=283 lang=golang
 *
 * [283] 移动零
 */

// @lc code=start
func moveZeroes(nums []int) {
	size := len(nums)
	// slow index
	lastNotZeroIndex := 0
	for i := 0; i < size; i++ {
		if nums[i] != 0 {
			nums[lastNotZeroIndex], nums[i] = nums[i], nums[lastNotZeroIndex]
			lastNotZeroIndex++
		}
	}
}

// @lc code=end

File: java/283movezores/283.移动零.java
/*
 * @lc app=leetcode.cn id=283 lang=java
 *
 * [283] 移动零
 */

// @lc code=start
class Solution {
    public void moveZeroes(int[] nums) {
        if (nums == null || nums.length < 2) {
            return;
        }
        int fastIndex = 0;
        int slowIndex = 0;
        int len = nums.length;
        for (; fastIndex < len; fastIndex++) {
            if (nums[fastIndex] != 0) {
                // swap
                int temp = nums[slowIndex];
                nums[slowIndex] = nums[fastIndex];
                nums[fastIndex] = temp;
                slowIndex++;
            }
        }
    }
}
// @lc code=end

